graphics/blender/Portfile

```diff
@@ -215,6 +238,12 @@ configure.args-append   -DWITH_PYTHON_INSTALL=OFF
 # MacPorts) Python framework
 configure.args-append   -DWITH_PYTHON_FRAMEWORK=ON
 
+# By preventing this Blender port from copying Python into the final
+# application bundle, and instead having Blender invoke the Python
+# interpreter externally through a framework, we can avoid having
+# Blender be directly dependent on Python.
+license_noconflict-append python${py_ver_nodot}
```

Comment:
Blender is just as dependent on Python whether or not you copy Python into the application bundle. Either the GPL-licensed scripts using Python's openssl bindings are distributable or they aren't. As I said on the mailing list, it's a more difficult than usual question, and I think we need to seek more expert opinions from the FSF and the Blender foundation.

Reply:
According to [Blender's license webpage](https://www.blender.org/about/license/), in the "License details" section, they state that "All the components that together make Blender are compatible under the newer GNU GPL Version 3. That is also the license to use for any distribution of Blender binaries." In the paragraph above that, they explicitly mention Python as being one of its components.

Is that still not a clear enough statement that Blender considers all of its components to be compatible for binary distribution? Do we need even more than that?

Reply:
Which brings up another question: Should I be adding GPL-3 to the `license` line? Because that licensing webpage says that its source code is licensed as GPL-2+, but its binaries are licensed as GPL-3.

Reply:
Based on this sentence on their licensing page:

> The source code we develop at blender.org is default being licensed as GNU GPL Version 2 or later

I believe `GPL-2+` is the correct thing to use in the portfile's license line.

Reply:
> Is that still not a clear enough statement that Blender considers all of its components to be compatible for binary distribution? Do we need even more than that?

The terms of the GPL are also very clear. I'd be very interested to know why they think they aren't a problem here. Who knows, maybe they haven't even thought about it.

Reply:
> Based on this sentence on their licensing page:
> 
> > The source code we develop at blender.org is default being licensed as GNU GPL Version 2 or later
> 
> I believe `GPL-2+` is the correct thing to use in the portfile's license line.

It does also say "the Blender Cycles rendering engine is available as Apache 2.0" though, and Apache-2 is not compatible with GPL-2. So if that component is included, the overall package could only be distributed under the terms of GPL-3.

Reply:
> > Is that still not a clear enough statement that Blender considers all of its components to be compatible for binary distribution? Do we need even more than that?
> 
> The terms of the GPL are also very clear. I'd be very interested to know why they think they aren't a problem here. Who knows, maybe they haven't even thought about it.

I can certainly e-mail the Foundation and ask... that's no problem at all. The only reason why I might sound hesitant is because they don't typically reply to these kinds of inquiries.

> >   Based on this sentence on their licensing page:
> > 
> > > The source code we develop at blender.org is default being licensed as GNU GPL Version 2 or later
> > 
> > I believe GPL-2+ is the correct thing to use in the portfile's license line.
> 
> It does also say "the Blender Cycles rendering engine is available as Apache 2.0" though, and Apache-2 is not compatible with GPL-2. So if that component is included, the overall package could only be distributed under the terms of GPL-3.

Yes, Blender Cycles is definitely included as a component. Cycles, along with EEVEE, are the two main rendering engines that Blender uses. EEVEE is the real-time renderer that is used in the GUI window. Cycles is the higher-quality but slower renderer that is mainly used for rendering final products. Wouldn't this seem to imply that I need to add `GPL-3` to the `license` line? Otherwise, how would we be able to indicate that the overall package was being distributed under GPL-3?

Reply:
So, I haven't sent out an e-mail to the Blender Foundation yet, but in the process of writing up my mail, I ran across this post/article written by Richard Stallman:

https://www.gnu.org/licenses/license-compatibility.en.html

I think one quote that is very relevant to the questions that are being asked both in this PR and on the mailing list, is this:

> The FSF uses the approach of asking people to release programs under “GNU GPL version N or any later version.” This licensing is compatible with version N, and also with N+1 (because it offers version N+1 as an option). When you combine code under “GPL 3 or later” with code under “GPL 2 or later”, the license of the combination is their intersection, which is “GPL 3 or later”.

Some excerpts from the post which are directly relevant to the Blender port:

> The Mozilla Public License permits relicensing to the GNU GPL except when the code explicitly denies this permission.

> The GNU GPL, version N, subsumes all versions of the Mozilla Public License that are compatible with it.

This would cover OpenVDB, which is licensed under MPL-2, since Blender is licensed under GPL 2 or later, which according to Stallman implies GPL-3 as well, and GPL-3 is compatible with MPL-2.

> Any GNU license, version N, subsumes the Apache 2.0 license provided N is at least 3.

This takes care of stuff like Blender Cycles, since "GPL v.2 or later" implies GPL-3, and thus is compatible with Apache 2.0. Stallman also classifies the Python license as being in the same category as Apache 2.0, which he calls a "lax permissive license":

> One lax license, Apache 2.0, has patent clauses which are incompatible with GPL version 2; since I think those patent clauses are good, I made GPL version 3 compatible with them.

After reading Stallman's post, I don't necessarily think it's the case that the folks at Blender Foundation haven't thought about this stuff yet. Instead, it seems like BF is of the belief that "GPL version 2 or later" automatically also means "GPL version 3", and GPL-3 is compatible with lax licenses such as Apache 2.0 and Python, and also compatible with MPL-2, EPL-1, etc. It would seem at this point that we not only have an indication from Blender Foundation, but also fairly definitive guidance from FSF itself, no?

Reply:
I agree that `Apache-2` + `GPL-2+` = `GPL-3`, and that that is the overall license of Blender itself. I said as much in my previous comment.

What I wanted clarification on is the GPL-licensed Python scripts that are using the OpenSSL bindings provided by the _hashlib and _ssl modules.

Reply:
E-mail has been sent to Blender Foundation. Awaiting reply.

Reply:
Joshua and I have received a reply from Blender Foundation. Just as Josh speculated,

> The terms of the GPL are also very clear. I'd be very interested to know why they think they aren't a problem here. Who knows, maybe they haven't even thought about it.

the Blender folks weren't aware of the license conflict. They've said that they will add licensing that will hopefully help to clarify their licensing terms, but those changes won't make it into Blender's official source code releases until version 2.91. So at least until then, we'll have to leave Blender as a port that is non-distributable as a binary.